src/util/gamma_picker.cpp

```diff
@@ -84,17 +89,27 @@ namespace lws
     static_assert(std::is_empty<crypto::random_device>(), "random_device is no longer cheap to construct");
     static constexpr const crypto::random_device engine{};
     const auto end = offsets().end() - CRYPTONOTE_DEFAULT_TX_SPENDABLE_AGE;
+    const uint64_t num_rct_outputs = spendable_upper_bound();
 
     for (unsigned tries = 0; tries < 100; ++tries)
     {
-      std::uint64_t output_index = std::exp(gamma(engine)) * outputs_per_second;
-      if (offsets().back() <= output_index)
+      double output_age_in_seconds = std::exp(gamma(engine));
+
+      // match wallet2 implementation to avoid creating distinct anonymity pools
+      // shift output back by unlock time to apply distribution from chain tip
+      if (output_age_in_seconds > defaut_unlock_time)
+        output_age_in_seconds -= defaut_unlock_time;
+      else
+        output_age_in_seconds = crypto::rand_idx(recent_spend_window);
```

Comment:
I regret not looking closely at `wallet2`. This answer is hopefully in that review ... why this? Its arbitrary and ruins the exponential selection.